docs/adr/0001-composite-handler-design.md

```diff
@@ -0,0 +1,73 @@
+# Composite Handler Design
+
+- Status: proposed
+- Decisers:
+  - Daniel Grant
+  - Tom Crane
+  - Donald Gray
+- Date: 2021-11-03
+
+## Context and Problem Statement
+
+### Context
+
+A design for the implementation and delivery of a component of the DLCS
+which can process "composite" artifacts. Such artifacts typically take
+the form of a single object containing multiple embedded resources, each
+of which can be rasterized and treated as an individual resource within
+the DLCS.
+
+## Considered Options
+
+- Additional standalone component which consumes a DLCS ingest payload,
+  retrieves the object, rasterizes accordingly, and then invokes the
+  existing DLCS ingestion workflow.
```

Comment:
Could the standalone tool use cantaloupe to rasterize? That could make it easier if we wanted to switch or had to switch down the line. Or does that defeat the purpose of having it standalone?

Reply:
When we discussed this, that was tabled as an option, but I think in doing so we end up doing more work than we have to - deployment, configuration and maintenance of Cantaloupe, in addition to the development of this additional standalone component which simply proxies through to that image server.

Contrast this with implementing rasterization within the handler - which _should_ be a relatively simple task - and to me I don't think it makes sense to do both.

Reply:
Yeah - we _could_ do this, but I think we'll come back to it later when we have Cantaloupe as a part of the DLCS.

There is a related issue here: https://github.com/dlcs/protagonist/issues/30#issuecomment-721274027

...where Cantaloupe might be used at ingest time as well as at end-user-facing runtime, to generate thumbs from sources without having to move them from S3.